Blizzard.IO.RabbitMQ/Rpc/NetqRabbitRpcServer.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Blizzard.IO.Core.Rpc;
+using Blizzard.IO.RabbitMQ.Entities.Rpc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Blizzard.IO.RabbitMQ.Rpc
+{
+    public class NetqRabbitRpcServer : IRpcServer
+    {
+        private readonly INetqRpcRabbitConnection<Func<Type,object>> _netqRpcRabbitConnection;
+        private readonly ILogger _logger;
+
+        private IDisposable _respondHandler;
+
+        public NetqRabbitRpcServer(INetqRpcRabbitConnection<Func<Type,object>> netqRpcRabbitConnection, ILoggerFactory loggerFactory)
+        {
+            _netqRpcRabbitConnection = netqRpcRabbitConnection;
+            _logger = loggerFactory.CreateLogger(nameof(NetqRabbitRpcServer));
+            _respondHandler = null;
+        }
+
+        public void Respond<TRequest, TRespond>(Func<TRequest, TRespond> callback)
+            where TRequest : class
+            where TRespond : class
+        {
+            VerifyServerHasntStartedYet();
+            if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Abstract)
```

Comment:
Why were not using a resolver on the MessageType?

Reply:
cause there are only two types of objects in c# abstract and concrete. it seems to just make the code less readable (also we wouldnt want to inject a dictionary from `RpcMessageType` to `Action` it from outside of the class)